File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if(n==0) return true;
        int N=flowerbed.length;
        for (int i = 0; i < N; i++) {
            if(flowerbed[i]==0){
                if(((i ==0 || flowerbed[i-1]==0)?0:1)==0 && ((i==flowerbed.length-1 || flowerbed[i+1]==0)?0:1)==0) {
                    flowerbed[i]=1;
                    n--;
                }
            }
            if(n==0) return true;
        }
        return false;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        ff = [0] + flowerbed + [0] 
        for i in range(1,len(ff)-1):
            if ff[i-1] == ff[i] == ff[i+1] == 0:
                n -= 1 
                ff[i] = 1
            if n <= 0:
                return True
        return n<=0